### ISSUE TITLE
Changing check for update functionality

### ISSUE THREAD

**Issue by aaabidunique** [subhra74/snowflake#106]
**Is your feature request related to a problem? Please describe.**
Check for update currently open below page for update information,
https://subhra74.github.io/snowflake/check-update.html?v=someversion
Instead I think checking for update in  background and open browser only when an update is available will be the better approach.

**Describe the solution you'd like**
Check in background and open browser only in case of update is available or else show 
*No Update Available*

**Describe alternatives you've considered**
NA
**Additional context**
NA

FYI,
I can do the said changes if you agree with the request.

P.S
If you agree with the idea I will be needing a json data from url similar to check for update url consisting of few informations.

**Comment by subhra74** [comment]
Its a good idea, you can submit a PR for this. For update related data, GitHub api for release can be used.

**Comment by aaabidunique** [comment]
@subhra74 I need to parse github api response but for that I think I should add json library(gson or alternative) instead of manually handling, since it can be used in future json data handling.
Any suggestions?

**Comment by subhra74** [comment]
Perhaps it would be great if user get a notification in main screen if a new version is release, instead of opening the info screen

**Comment by aaabidunique** [comment]
Ya thats an excellent idea.
I think we should incorporate both the functionality.
What do you think??

**Comment by subhra74** [comment]
Yes, sounds great. For json you can use jackson, as its already been used in the project

**Comment by subhra74** [comment]
Also if you are implementing notifications, try not to use joptionpane or any modal dialog

**Comment by aaabidunique** [comment]
I don't have much experience with swing, so what do you suggest in place of joptionpane or model??

**Comment by subhra74** [comment]
Perhaps a small notification panel inside the main window. Or a tray notification could work as wel

**Comment by aaabidunique** [comment]
I think notification panel/ tray notification may work but since this project is available as other packages (snap/flatpak) also and which get updated via system update, we should think of different strategy.
Example : while opening the app we will perform check for update in background and if update is available, we will show a dialog with changelog and button to open the github page.

Suggestion : We should implement menus to club functionlity and maybe add more functionlity in future and for update we can add it in about menu.

**Comment by subhra74** [comment]
personally i do not like modal dialog opening when i am in a hurry and opening the app shows a modal and blocks the ui. notification is better, and when user clicks on the notification, the dialog can be shown which can show change-log and if user is interested then he can click on the update button which might redirect to github. if user is using snap or some other distro specific update, we can introduce a command line flag to not to check for update. the command line can be added to the launcher shell script

**Comment by aaabidunique** [comment]
A slightly different approach.

- We will check for update on app startup.
- If update is available show update button in topbar.
- clicking update button (topbar) will show update dialog with changelog (if available).
- clicking update in dialog will open browser and land on latest release tag page.

![Screenshot from 2020-03-30 11-24-13](https://user-images.githubusercontent.com/9800757/77880229-c6320e80-7279-11ea-8334-e84d13755e6d.png)
![Screenshot from 2020-03-30 11-24-20](https://user-images.githubusercontent.com/9800757/77880231-c7fbd200-7279-11ea-801a-8d77c454f1e6.png)

Any feedback on the approach?

**Comment by aaabidunique** [comment]
@subhra74 Any comments for the above?

**Comment by subhra74** [comment]
thanks, it looks good. How ever if you are willing to implement this please make changes to muon branch, as it will become master soon. There is already a button, so you can just add some badge to point out new update is available:
![image](https://user-images.githubusercontent.com/8403090/80465184-27153b00-893b-11ea-8be2-b30df95396ff.png)

**Comment by aaabidunique** [comment]
@subhra74 I think it will be better once you move muon to master and everything is working fine, I will make the changes to master.
